### PR TITLE
Reopen "Change test_name's formart to file_path#class#func"

### PR DESCRIPTION
**Pull request by ninjinkun**
@shuheiktgw  I reopen https://github.com/launchableinc/nose-launchable/pull/10. Sorry for confusing you.

### CODE REVIEW

**Comment by shuheiktgw** [comment]
@ninjinkun https://github.com/launchableinc/nose-launchable/pull/10 I left a few comments on the PR! 🙏

**Comment by ninjinkun** [comment]
@shuheiktgw I continue #10 discussion on this PR.

https://github.com/launchableinc/nose-launchable/pull/10#issuecomment-716257900 Yes. I worried about the same thing. Developers change directories not frequently but sometimes they do. Although we can add an option to specify the root directory, I worried they can't continue to maintain the relationship between the root directory and test directories.

**Comment by ninjinkun** [comment]
I'm researching how Git solves this consistency problem

**Comment by ninjinkun** [comment]
https://github.com/launchableinc/nose-launchable/pull/10#issuecomment-716256383 I fixed this issue.

**Comment by ninjinkun** [comment]
https://github.com/launchableinc/nose-launchable/pull/10#issuecomment-716258487 For this test generator problem, I plan to aggregate all generated tests' results. For example, `test_evens(0, 0)` is succeeded, `test_evens(1, 1)` is failed, `test_evens(2, 2)` is succeeded, this plugin sends a `test_evens` failed event. As a negative point, It needs waiting for all tests are done because we cannot know when all generated tests are done.

**Comment by shuheiktgw** [comment]
@ninjinkun Hmm, `test.id()` seems to have a generated test name can we use it to identify a test name?

```
(Pdb) test.id()
'tests.dir3.test1.test_evens(0, 0)'
```

**Comment by ninjinkun** [comment]
@shuheiktgw It works. However, I worried about generating tests with random parameters. If we use `test.id()`, we need to group these tests on the server.

**Comment by ninjinkun** [comment]
@shuheiktgw But at this time, we need to go forward. I'm going to use the name with parameters such as `test_evens(0, 0)`.

File: launchable/plugin.py [hunk 1]
```diff
@@ -135,12 +136,13 @@ def _getCapturedStderr(self):
         return ''
 
     def _addResult(self, test, status, queueing):
+        # return such as tests/dir1/test1.py#test1#test_evens
         def get_test_name(t):
-            if isinstance(t.context, ModuleType):
-                return t.context.__name__
+            file_path, module, = test_address(t.test)
```
Comment (shuheiktgw) [inline launchable/plugin.py:141]: I guess a function name is missing?

File: launchable/plugin.py [hunk 2]
```diff
@@ -92,7 +92,8 @@ def afterTest(self, test):
 
     @protect
     def addError(self, test, err, capt=None):
-        self._addResult(test, CaseEvent.TEAT_FAILED, self._uploader.enqueue_failure)
+        if err is not ImportError:
```
Comment (shuheiktgw) [inline launchable/plugin.py:95]: It seems if a non-existing test is specified, err is tuple like: `(<class 'ValueError'>, ValueError('No such test aaa'), <traceback object at 0x1044a2040>)`. You can check this by running `nosetests tests/dir3/test1.py:aaa -v --nocapture --nologcapture --launchable --launchable-build-number 1` in the rocket-car repo!

**Comment by ninjinkun** [comment]
@shuheiktgw My apologies, my local testing has failed because of the wrong auth token. I'm going to fix my local, then fix my code.

File: launchable/plugin.py (hunk 1 above)
Comment (ninjinkun) [reply launchable/plugin.py]: I fixed it!

File: launchable/plugin.py (hunk 2 above)
Comment (ninjinkun) [reply launchable/plugin.py]: I fixed to handle ValueError type

**Comment by ninjinkun** [comment]
I fixed them

**Review by shuheiktgw — APPROVED** [review]
LGTM! 👍

**Pull request merged**